enhancements/cluster-logging/hypershift-audit-logs.md

```diff
@@ -0,0 +1,249 @@
+- reconciled by the _hypershift logging operator_ on the _management_ cluster.
+- contains the service provider's forwarding configuration, to be applied to every HCP.
+  - combined with service consumer's configuration if it exists
+
+### Implementation Details
+
+#### Hypershift Logging Operator
+
+The _Hypershift Logging Operator_ (HLO) runs on the management cluster.
+
+**NOTE**: Unlike most operators, it watches resources on both the control plane and the data plane.
+
+- Control plane 
+  - HCP namespaces (goal is to enable logging in each of these)
+  - `HypershiftLogWatcher` (service provider logging configuration)
+- Data plane
+  - `HypershiftLogForwarder` (service consumer logging configuration)
+  - `Policy.audit.k8s.io.` (service consumer audit policy)
+
+Reconciliation:
+
+- Validate `HypershiftLogForwarder` resource in the data plane
+- Copy resources referred to by the HLF from data to control plane:
+  - `Policy` resources for audit policies
+  - `Secret` and `ConfigMap` resources for log outputs
+- Copy/generate non-resource credentials required by the HLF
+- Update `ClusterLogForwarder` resource in the HCP namespace (control plane) combining:
+  - HLF configuration from the data plane (if present)
+  - CLF template from the `HypershiftLogWatcher` from the control plane (if present)
+
+The result is a CLF resource in each HCP namespace that combines valid service consumer configuration (from data plane)
+with service provider configuration (from the control plane).
+
+The CLO manages these resources as normal, by deploying collectors to forward logs.
+
+#### Cluster Logging Operator
+
+Two new CLO features are needed:
+
+- **Multiple instances**: The ClusterLogForwarder is currently a singleton. \
+  To provide isolation between HCPs we need to deploy a forwarder per HCP namespace.
+  This requires modifications to the CLO.
+  See [LOG-1343 Multiple forwarders](https://issues.redhat.com/browse/LOG-1343).
+
+- **HTTP (aka webhook) server**: The ClusterLogForwarder needs to be extended to act a a HTTP listener,
+  so it can receive logs as a web hook from the API server in the HCP.
+  See [LOG-3965 Collector to act as http server](https://issues.redhat.com/browse/LOG-3965)
+
+### Risks and Mitigations
+
+**NOTE**: See also "Risks and Mitigations" in [api-audit-log-policy](/api-audit-log-policy.md)
+
+### Drawbacks
+
+## Design Details
+
+### Open Questions
+
+Need to ensure the management cluster can get credentials to connect to log outputs.
+Credentials in Secrets are not a problem, but credentials that are magically auto-mounted in various ways may be tricky.
+Need to review security requirements of existing outputs.
+
+### Test Plan
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+#### Tech Preview -> GA
+
+#### Removing a deprecated feature
+
+None.
+
+### Upgrade / Downgrade Strategy
+
+Upgrading: Existing customer `cl/instance` and `clf/instance` resources will not be modified.
```

Comment:
How can this be true if CLF is moving from global to namespace scope?

Reply:
The CLF is already in namespace scope, but restricted to only deploy in the "openshift-logging" namespace with the name "instance", so there can be only one instance. We just need to lift that restriction and fix assumptions in the code about the constant namespace and name.